Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift

```diff
@@ -265,43 +264,42 @@ public struct MessageContainerView<Factory: ViewFactory>: View {
                         : nil
                 )
 
-                if !message.isRightAligned {
+                if !messageViewModel.isRightAligned {
                     MessageSpacer(spacerWidth: spacerWidth)
                 }
             }
         }
         .padding(
             .top,
-            topReactionsShown && !isMessagePinned ? messageListConfig.messageDisplayOptions.reactionsTopPadding(message) : 0
+            topReactionsShown && !messageViewModel.isPinned ? messageListConfig.messageDisplayOptions
+                .reactionsTopPadding(message) : 0
         )
         .padding(.horizontal, messageListConfig.messagePaddings.horizontal)
-        .padding(.bottom, showsAllInfo || isMessagePinned ? paddingValue : 2)
+        .padding(.bottom, showsAllInfo || messageViewModel.isPinned ? paddingValue : 2)
         .padding(.top, isLast ? paddingValue : 0)
-        .background(isMessagePinned ? Color(colors.pinnedBackground) : nil)
-        .padding(.bottom, isMessagePinned ? paddingValue / 2 : 0)
+        .background(messageViewModel.isPinned ? Color(colors.pinnedBackground) : nil)
+        .padding(.bottom, messageViewModel.isPinned ? paddingValue / 2 : 0)
         .transition(
             message.isSentByCurrentUser ?
                 messageListConfig.messageDisplayOptions.currentUserMessageTransition :
                 messageListConfig.messageDisplayOptions.otherUserMessageTransition
         )
         .accessibilityElement(children: .contain)
         .accessibilityIdentifier("MessageContainerView")
+        .id(message.id + messageViewModel.textContent)
```

Comment:
This was added so that whenever the view model textContent changes, it should reset the view. This is useful for translations, markdown etc..

Reply:
hmm, this doesn't feel right. The view should re-render automatically when the text changes.

Reply:
Without this, it does not work, tho, the textContent is not updated. I think it is most likely due to how we have the LinkDetectionTextView

Reply:
This one is important to fix, since the explicit view being setup like this might prevent other updates (adding reactions, read counts etc). So try to figure it out, if you don't have luck, I will also have a go on this one.

Reply:
Discussed internally and now it is fixed 👍